test/powershell/Modules/Microsoft.PowerShell.Utility/Add-Type.Tests.ps1

```diff
@@ -207,6 +207,12 @@ public class AttributeTest$guid : PSCmdlet
         { Add-Type -CompilerOptions "/platform:anycpuERROR" -Language CSharp -MemberDefinition "public static string TestString() { return ""}" -Name "TestType1" -Namespace "TestNS" -ErrorAction Stop } | Should -Throw -ErrorId "SOURCE_CODE_ERROR,Microsoft.PowerShell.Commands.AddTypeCommand"
     }
 
+    It "Throw if the type already exists" {
+        Add-Type -TypeDefinition "public class Foo$guid {}"
+        { Add-Type -TypeDefinition "public class Foo$guid { public int Bar {get {return 42;} }" -ErrorAction SilentlyContinue } | Should -Throw -ErrorId "COMPILER_ERRORS,Microsoft.PowerShell.Commands.AddTypeCommand"
+        $error[1].FullyQualifiedErrorId | Should -BeExactly "TYPE_ALREADY_EXISTS,Microsoft.PowerShell.Commands.AddTypeCommand"
```

Comment:
You should use `-PassThru` to check the ErrorRecord rather than relying on `$error[1]`.

Reply:
Fixed.

Reply:
I see.  Perhaps add a comment.

Reply:
Comment added.